### DISCUSSION TITLE
Is it possible to use this library with STM32CubeIDE?

### DISCUSSION THREAD

**Discussion by rtek1000** [hathach/tinyusb#1827] [Q&A]
Hello,

I'm having a little trouble implementing USB Host HID with HUB. The mini keyboard with built-in touchpad has an internal HUB, so I need HUB support, but currently STM32CubeIDE does not have HUB support.

![image](https://user-images.githubusercontent.com/8660811/209983630-c3f7987d-d216-4981-be55-1e7e4290b838.png)

**Comment by rtek1000** [comment]
Another alternative is the Arduino core for STM32, but the developers told me that the core doesn't support this library. 

https://github.com/stm32duino/Arduino_Core_STM32/discussions/1912#discussioncomment-4548837

**Comment by hathach** [comment]
any IDE that support gcc should compile fine with this lib, though you have to do all the glues

**Comment by kf6gpe** [comment]
Yes, we successfully used it with a CubeMX project for the STM32H7. You may want to look at family.c for your specific MCU & board to make sure you get the H/W successfully configured.
